pkg/apis/core/v1beta1/types_namespacedcloudprofile.go

```diff
@@ -0,0 +1,98 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NamespacedCloudProfile represents certain properties about a provider environment.
+type NamespacedCloudProfile struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Spec defines the provider environment properties.
+	Spec NamespacedCloudProfileSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	// Most recently observed status of the NamespacedCloudProfile.
+	Status NamespacedCloudProfileStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NamespacedCloudProfileList is a collection of CloudProfiles.
+type NamespacedCloudProfileList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list object metadata.
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Items is the list of NamespacedCloudProfiles.
+	Items []NamespacedCloudProfile `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// NamespacedCloudProfileSpec is the specification of a NamespacedCloudProfile.
+type NamespacedCloudProfileSpec struct {
+	// CABundle is a certificate bundle which will be installed onto every host machine of shoot cluster targeting this profile.
+	// +optional
+	CABundle *string `json:"caBundle,omitempty" protobuf:"bytes,1,opt,name=caBundle"`
+	// Kubernetes contains constraints regarding allowed values of the 'kubernetes' block in the Shoot specification.
+	// +optional
+	Kubernetes *KubernetesSettings `json:"kubernetes,omitempty" protobuf:"bytes,2,opt,name=kubernetes"`
+	// MachineImages contains constraints regarding allowed values for machine images in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	MachineImages []MachineImage `json:"machineImages,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,3,opt,name=machineImages"`
+	// MachineTypes contains constraints regarding allowed values for machine types in the 'workers' block in the Shoot specification.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	MachineTypes []MachineType `json:"machineTypes" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,4,opt,name=machineTypes"`
+	// Regions contains constraints regarding allowed values for regions and zones.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	Regions []Region `json:"regions" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,6,opt,name=regions"`
```

Comment:
protobuf number 5 is missing, and similary profobuf number 8 is missing. I am assuming you have copied protobuf from `Cloudprofie`.
We don't have to write protobuf on our own and it should be generated automatically.
Please removed `protobuf:<*>`  from all the newly added fields and the run `make generate`, protobuf will get automatically populated.

Reply:
Thanks for the info, protobuf config for the type should be correct in https://github.com/gardener/gardener/pull/9440/commits/026d5bdc62ce06a524e6839f6d663a7dcf36a6e6